Document().to_json() is not JSON serializable

### Software versions

Python version        :  3.12.12 (main, Jan  7 2026, 13:39:36) [Clang 17.0.0 (clang-1700.6.3.2)]
IPython version       :  8.23.0
Tornado version       :  6.5.1
NumPy version         :  1.26.4
Bokeh version         :  3.9.0
BokehJS static path   :  ~/.pyenv/versions/3.12.12/envs/dev@3.12.12/lib/python3.12/site-packages/bokeh/server/static
node.js version       :  v18.4.0
npm version           :  8.12.1
jupyter_bokeh version :  (not installed)
Operating system      :  macOS-26.3.1-arm64-arm-64bit

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

Expected `print(json.dumps(Document().to_json()))` to work (works in Bokeh 3.8.2).

### Observed behavior

```
    print(json.dumps(Document().to_json()))
../../.pyenv/versions/3.12.12/lib/python3.12/json/__init__.py:231: in dumps
    return _default_encoder.encode(obj)
../../.pyenv/versions/3.12.12/lib/python3.12/json/encoder.py:200: in encode
    chunks = self.iterencode(o, _one_shot=True)
../../.pyenv/versions/3.12.12/lib/python3.12/json/encoder.py:258: in iterencode
    return _iterencode(o, 0)
../../.pyenv/versions/3.12.12/lib/python3.12/json/encoder.py:180: in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
E   TypeError: Object of type Serialized is not JSON serializable
```

### Example code

```Python
print(json.dumps(Document().to_json()))
```

### Stack traceback or browser console output

_No response_

### Screenshots

_No response_

--- a/src/bokeh/document/document.py
+++ b/src/bokeh/document/document.py
@@ -772,12 +772,26 @@ side of a communications channel while it was being removed on the other end.\
     def to_json(self, *, deferred: Literal[False]) -> DocJson: ...
 
     def to_json(self, *, deferred: bool = True) -> DocJson | Serialized[DocJson]:
-        ''' Convert this document to a JSON-serializable object.
+        ''' Convert this document to a serialized representation.
+
+        .. note::
+            Despite the name, the default return value is **not** directly
+            JSON-serializable. With ``deferred=True`` (the default), any binary
+            buffers are kept as references and the result is a ``Serialized``
+            wrapper, so e.g. ``json.dumps(doc.to_json())`` will raise
+            ``TypeError``. To obtain a JSON *string*, pass the result to
+            ``bokeh.core.json_encoder.serialize_json``. Alternatively, pass
+            ``deferred=False`` to inline any binary buffers as base64 and return
+            a plain ``dict`` that is directly JSON-serializable.
 
         Args:
-            deferred (bool) : encode buffers lazily as references or immediately as inline (base64)
+            deferred (bool) :
+                If ``True`` (default), encode binary buffers lazily as
+                references and return a ``Serialized`` wrapper. If ``False``,
+                encode buffers immediately as inline base64 and return a plain
+                ``DocJson`` dict that is directly JSON-serializable.
 
-        Return:
+        Returns:
             Serialized[DocJson] | DocJson
 
         '''

